qemu/tests/virtio_driver_sign_check.py

```diff
@@ -1,87 +1,153 @@
+        """
+        guest_name = params["guest_name"]
+        guest_list = dict([x.split(":") for x in alias_map.split(",")])
+        os_arch_path = guest_list[guest_name]
+        global os_arch_path_list
+        os_arch_path_list = os_arch_path.split("\\")
+        return os_arch_path_list
+
+    def filter_all_tested_cat_files(cat_files):
+        """
+        In all suffix '.cat' files, select that will be tested '.cat' file.
+
+        :param cat_files:all suffix '.cat' files.
+        :return True:correct os path, arch path and the dirver_dir
+        belong in need_test_driver list, return True.
+        """
+        pattern = r'(\w:)\\(.+?)\\(.+?)\\(.+?)[\.|\\]'
+        ret = re.match(pattern, cat_files, re.I)
+        disk_name = ret.group(1)
+        os_name = ret.group(3)
+
+        if disk_name == "A:":
+            get_os_arch_path(alias_map=params.get("guest_alias_fl"))
+            os_path = os_arch_path_list[1]
+            arch_path = os_arch_path_list[0]
+            plat_name = ret.group(2)
+            driver_dir = ret.group(4)
+        else:
+            get_os_arch_path(alias_map=params.get("guest_alias"))
+            os_path = os_arch_path_list[0]
+            arch_path = os_arch_path_list[1]
+            driver_dir = ret.group(2)
+            plat_name = ret.group(4)
+
+        if (os_name == os_path) and (plat_name == arch_path):
+            if driver_dir in need_test_driver:
+                return True
+
+    def match_cat_and_verified_file(filterd_cat_file):
+        """
+        Tuple each selected '.cat' file and with same directory '.sys' file;
+        Tuple each selected '.cat' file and with same directory '.inf' file;
+        Tuple each selected '.cat' file and with same directory 'Wdf' file.
+
+        Collect all tuples in virtio_drivers list to be used signtool test.
+
+        :param filterd_cat_file:each filterd suffix '.cat' file
+        """
+        pattern = r'\w:\\(.+?)\\(.+?)\\(.+?)[\.|\\]'
+        match_ret = re.match(pattern, filterd_cat_file)
+        driver_dir = match_ret.group()
+
+        driver_sys = filter(lambda x: driver_dir in x, drivers['.sys'])
+        driver_inf = filter(lambda x: driver_dir in x, drivers['.inf'])
+        driver_wdf = filter(lambda x: driver_dir in x, drivers['Wdf'])
+
+        cat_sys = map(lambda x: (filterd_cat_file, x), driver_sys)
+        cat_inf = map(lambda x: (filterd_cat_file, x), driver_inf)
+        cat_wdf = map(lambda x: (filterd_cat_file, x), driver_wdf)
+        cat_parm = cat_sys + cat_inf + cat_wdf
+        virtio_drivers.extend(cat_parm)
+
     try:
-        for drive in drive_list.split():
-            for type in ['.cat', '.sys']:
+        error_context.context("Running SignTool check test in guest...",
+                              logging.info)
+        disk_result = session.cmd_output(get_disk_letter)
+        disk_letter = re.search(r'(\w:)\s', disk_result, re.M).group(1)
```

Comment:
utils_misc.get_win_disk_vol() can get virtio-win disk letter.

Reply:
try to optimize here, thanks~